### PR TITLE
refactor: adopt account discovery

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
This PR is to adopt the `AccountService` into `AccountRpcController` in best effort, but not to revamp the RPC/other utils to make use of the `Account` object

### Changes:
- Fix `AccountContractDiscovery` to select cairo1 contract, in case more than 1 contract deployed.
- Fix `CairoAccountContract` to return correct value for `isRequireDeploy` method
- Adopt `AccountService` in `AccountRpcController`
- 2 RPCs are impacted and updated with min change
  - `starkNet_executeTxn` -> update address index name from the `Account` object
  - `starkNet_getAddrFromStarkName` -> update parent controller

- Rename some test utils prefix from "prepareXXX" to "setupXXXXTest"
- Fix the tests for RPC

### Requirements:
- [ ] #472

### CODE REVIEW

**Review by khanti42 — CHANGES_REQUESTED** [review]
Just a very minor comment on making the comments of account selection clearer on edge cases scenario

File: packages/starknet-snap/src/wallet/account/discovery.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export class AccountContractDiscovery {
    * 1. If a contract is deployed, then use the deployed contract.
    * 2. If no contract is deployed, but has balance, then use the contract with balance.
    * 3. If neither contract is deployed or has balance, then use the default contract.
+   * 4. If multiple contracts are deployed, then use the default contract.
```
[the remark this answers is not in the document]
Comment (stanleyyconsensys) [reply packages/starknet-snap/src/wallet/account/discovery.ts]: i refine the text in other way, please see if it is clear

**Review by khanti42 — APPROVED** [review]
